pegen/pegen.c

```diff
@@ -30,12 +41,13 @@ raise_syntax_error(Parser *p, const char *errmsg, ...)
     } else {
         Py_INCREF(Py_None);
         filename = Py_None;
-        loc = PyUnicode_DecodeUTF8(p->tok->buf, (p->tok->cur)-(p->tok->buf), "replace");
+        loc = PyUnicode_FromString(p->tok->buf);
         if (!loc) {
             goto error;
         }
     }
-    tmp = Py_BuildValue("(OiiN)", filename, t->lineno, t->col_offset + 1, loc);
+    Py_ssize_t col_offset = byte_offset_to_character_offset(loc, t->col_offset + 1);
```

Comment:
This should add 1 to the result of the call, not to the input -- if there's a multibyte character at the offset, the +1 will cause PyUnicode_DecodeUTF8 to fail. And I'd call the local variable col_number, since it's 1-based.